alternator/streams.cc

```diff
@@ -145,19 +145,24 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
     auto table = find_table(_proxy, request);
     auto db = _proxy.data_dictionary();
     auto cfs = db.get_tables();
-    auto i = cfs.begin();
-    auto e = cfs.end();
 
     if (limit < 1) {
         throw api_error::validation("Limit must be 1 or more");
     }
 
-    // TODO: the unordered_map here is not really well suited for partial
-    // querying - we're sorting on local hash order, and creating a table
-    // between queries may or may not miss info. But that should be rare,
-    // and we can probably expect this to be a single call.
+    // # 12601 (maybe?) - sort the set of tables on ID. This should ensure we never
+    // generate duplicates in a paged listing here. Can obviously miss things if they 
+    // are added between paged calls and end up with a "smaller" UUID/ARN, but that 
+    // is to be expected.
```

Comment:
I don't think that "duplicates" is necessarily the main issue here. I think the point is that the list of tables may be sorted differently at different times, so we can't continue "after" some id X in the next if the meaning of "after" changed between pages. We can get the same result more than once, and not get other results (we saw this in 12601). We can also get the wrong number of results. In fact, I'm a bit worried that in 12601 we only saw a small error - it could have been much worse (with many duplicates and/or missing items), so I hope this is really this bug and not something else.

Reply:
Duplicates in the sense that we report a stream twice, because relative position of "last" changes. Obviously can also miss entries even though they existed before we started paging. But that is "better" in the sense that we are still showing a coherent set (and given that streams. Still not good of course. 
Also hope this fixes the issue. Hard to know.